File: javatest/omnia/data/cache/CachedLongTest.java
package omnia.data.cache;

import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class CachedLongTest {

  @Test
  public void cache_didReturnSimpleLongCache() {
    assertTrue(CachedLong.cache(() -> 132L) instanceof SimpleLongCacher);
  }
}

File: omnia/src/commonTest/omnia/data/cache/CachedLongTest.kt
package omnia.data.cache

import kotlin.test.Test
import omnia.util.test.fluent.Assertion.Companion.assertThat
import omnia.util.test.fluent.isA

class CachedLongTest {

  @Test
  fun cache_didReturnSimpleLongCache() {
    assertThat(CachedLong.cache { 132L }).isA(SimpleLongCacher::class)
  }
}